### PR TITLE
docs: add jina 3 migration guide

### PR DESCRIPTION
**Pull request by JohannesMessner**
# Jina 3 migration guide

Add Jina 3 migration guide

### CODE REVIEW

**Review by JoanFM — COMMENTED** [review]
Remember to do the PR to `docs3.0` branch

File: docs/migrate-to-jina3/migrate.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Migration to Jina 3
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:1]: Are we going with `3` or `3.0`?

File: docs/migrate-to-jina3/migrate.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+## Attribute renamings
+
+Docarray introduces more natural naming conventions for `Document` and `DocumentArray` attributes.
+
+- `doc.blob` is renamed to `doc.tensor`, to align with external libraries like PyTorch and Tensorflow
+- `doc.buffer` is renamed to `doc.blob`, to align with the industry standard
+
+## Accessing attributes and elements
+
+Docarray introduces a flexible way of accessing attributes of `Document`s in a `DocuemntArray`, in bulk.
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:19]:
```suggestion
Docarray introduces a flexible way of bulk-accessing attributes of `Document`s in a `DocumentArray`.
```

File: docs/migrate-to-jina3/migrate.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- `docs.flatten()` is replaced by `docs[...]`
+
+Batching operations are delegated to the docarray package and Python builtins:
+
+- `docsarray.batch()` does not accept the arguments `traversal_paths=` and `require_attr=` anymore.
+To achieve previously allowed behaviours like `docs.batch(traversal_paths=paths, batch_size=bs, require_attr='attr')`
+use the simplified `.batch()` method in combination with Python's `filter()`:
+`DocumentArray(filter(lambda x : bool(x.attr), docs[path])).batch(batch_size=bs)`
+
+## Method behaviour changes
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:44]:
```suggestion
## Method behavior changes
```

We stick to US-English by default

File: docs/migrate-to-jina3/migrate.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+compared to Jina 2, exposing entire `Response`s to the user. By setting or unsetting the `results_as_docarray=` flag,
+the user can override these default behaviours.
+- In Jina 2, bulk accessing attributes in a `DocumentArray` returns a list of empty values, when the `Document`s
+inside the `DocumentArray` do not have a value for that attribute. In Jina 3, this returns `None`. This change becomes
+important when migrating code that checks for the presence of a certain attribute.
+
+
+## Serialization
+
+- `docs.SerializeToString()` is removed in favour of `doc.to_bytes()`
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:57]: `doc` or `docs`?

Comment (alexcg1) [reply docs/migrate-to-jina3/migrate.md:57]: Both are mentioned

File: docs/migrate-to-jina3/migrate.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+- Creating a `Document` from serialized data using `Document(bytes)` is removed in favour of
+`Document.from_bytes(bytes)`
+
+
+## YAML parsing
+
+In Jina 3, yaml syntax is aligned with [github actions notation](https://docs.github.com/en/actions/learn-github-actions/environment-variables),
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:64]:
```suggestion
In Jina 3, YAML syntax is aligned with [GitHub Actions notation](https://docs.github.com/en/actions/learn-github-actions/environment-variables),
```

File: docs/migrate-to-jina3/migrate.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+which leads to the following changes:
+
+- Referencing environment variables using the syntax `${{ VAR }}` is no longer allowed. The posix notations for
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:67]:
```suggestion
- Referencing environment variables using the syntax `${{ VAR }}` is no longer allowed. The POSIX notations for
```

File: docs/migrate-to-jina3/migrate.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Migration to Jina 3
+
+Jina 3 introduces a number of exciting features, but to be able to enjoy them, you will also have to make some
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:3]: Do we want to mention (and link to) some of the new features briefly? That'll get people excited to use 3.0 and will want to read the rest of this guide

File: docs/migrate-to-jina3/migrate.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+tweaks to your existing Jina 2 code.
+
+One of the major changes in Jina 3 is the inclusion of the [docarray library](https://docarray.jina.ai/).
+Accordingly, the scope of the changes in Jina 3 will be mainly related to `Document` and `DocumentArray`.
+Here you can find out what exactly you will have to change.
+
+## Attribute renamings
```
Comment (alexcg1) [inline docs/migrate-to-jina3/migrate.md:10]: Could we rename some of these titles to highlight the benefit of the change?

e.g. `More natural attribute names`

Not sure we can do it for every title, but let's give it a shot

**Review by alexcg1 — CHANGES_REQUESTED** [review]

File: docs/migrate-to-jina3/migrate.md (hunk 1 above)
Comment (JohannesMessner) [reply docs/migrate-to-jina3/migrate.md:1]: I think 3 is better, assuming that we don't have more breaking changes any time too soon

File: docs/migrate-to-jina3/migrate.md (hunk 7 above)
Comment (JohannesMessner) [reply docs/migrate-to-jina3/migrate.md:3]: Definitely, but I don't know that we have such a link yet

**Comment by JohannesMessner** [comment]
Does anyone know how to manually trigger the netlify deployment?

**Pull request closed without merging**